pkg/proxy/topology_test.go

```diff
@@ -0,0 +1,464 @@
+			// zone, catch-all ("*") matched with 8 endpoints
+			Name: "multiple topology keys matched node with 'Any' key",
+			nodeLabels: map[types.NodeName]map[string]string{
+				"testNode1": {
+					"kubernetes.io/hostname":                   "testNode1",
+					"failure-domain.beta.kubernetes.io/zone":   "90001",
+					"failure-domain.beta.kubernetes.io/region": "cd",
+				},
+				"testNode2": {
+					"kubernetes.io/hostname":                   "testNode2",
+					"failure-domain.beta.kubernetes.io/zone":   "90001",
+					"failure-domain.beta.kubernetes.io/region": "cd",
+				},
+				"testNode3": {
+					"kubernetes.io/hostname":                   "testNode3",
+					"failure-domain.beta.kubernetes.io/zone":   "90002",
+					"failure-domain.beta.kubernetes.io/region": "cd",
+				},
+				"testNode4": {
+					"kubernetes.io/hostname":                   "testNode4",
+					"failure-domain.beta.kubernetes.io/zone":   "90002",
+					"failure-domain.beta.kubernetes.io/region": "cd",
+				},
+				"testNode5": {
+					"kubernetes.io/hostname":                   "testNode5",
+					"failure-domain.beta.kubernetes.io/zone":   "90003",
+					"failure-domain.beta.kubernetes.io/region": "cd",
+				},
+			},
+			endpoints: []Endpoint{
+				&BaseEndpointInfo{Endpoint: "1.1.1.1:11", NodeName: "testNode1"},
+				&BaseEndpointInfo{Endpoint: "1.1.1.2:11", NodeName: "testNode1"},
+				&BaseEndpointInfo{Endpoint: "1.1.2.1:11", NodeName: "testNode2"},
+				&BaseEndpointInfo{Endpoint: "1.1.2.2:11", NodeName: "testNode2"},
+				&BaseEndpointInfo{Endpoint: "1.1.3.1:11", NodeName: "testNode3"},
+				&BaseEndpointInfo{Endpoint: "1.1.3.2:11", NodeName: "testNode3"},
+				&BaseEndpointInfo{Endpoint: "1.1.4.1:11", NodeName: "testNode4"},
+				&BaseEndpointInfo{Endpoint: "1.1.4.2:11", NodeName: "testNode4"},
+			},
+			currentNodeName: "testNode5",
+			topologyKeys:    []string{"kubernetes.io/hostname", "failure-domain.beta.kubernetes.io/zone", v1.TopologyKeyAny},
+			expected: []Endpoint{
+				&BaseEndpointInfo{Endpoint: "1.1.1.1:11", NodeName: "testNode1"},
+				&BaseEndpointInfo{Endpoint: "1.1.1.2:11", NodeName: "testNode1"},
+				&BaseEndpointInfo{Endpoint: "1.1.2.1:11", NodeName: "testNode2"},
+				&BaseEndpointInfo{Endpoint: "1.1.2.2:11", NodeName: "testNode2"},
+				&BaseEndpointInfo{Endpoint: "1.1.3.1:11", NodeName: "testNode3"},
+				&BaseEndpointInfo{Endpoint: "1.1.3.2:11", NodeName: "testNode3"},
+				&BaseEndpointInfo{Endpoint: "1.1.4.1:11", NodeName: "testNode4"},
+				&BaseEndpointInfo{Endpoint: "1.1.4.2:11", NodeName: "testNode4"},
+			},
+		},
+	}
+	for tci, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			filteredEndpoint := FilterTopologyEndpoint(tc.currentNodeName, tc.nodeLabels, tc.topologyKeys, tc.endpoints)
+			if !reflect.DeepEqual(filteredEndpoint, tc.expected) {
+				t.Errorf("[%d] expected %v, got %v", tci, endpointsToStringArray(tc.expected), endpointsToStringArray(filteredEndpoint))
```

Comment:
`[%d]` is not needed, t.Run will print the test name on error

Reply:
done